### PR TITLE
Dev/wip 1139 title progress

### PR DESCRIPTION
**Pull request by SuttArt**
see issue: #1139

### Frontend Changes
- [x] Add reactive form control for Title in "Progress Dashboard widget"

### Backend Changes
- [x] Send FormGroup from edit-progress-dashboard.component.ts with ProgressDashboardConfig.title and ProgressDashboardConfig.parts
- [x] Subscribe all data (title and parts) in progress-dashboard.component.ts

### CODE REVIEW

File: src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.html [hunk 1]
```diff
@@ -92,8 +100,8 @@ <h1 i18n="Edit the progress of one or multiple tasks">Edit Progress</h1>
 <mat-dialog-actions>
     <button
       mat-stroked-button
-      [mat-dialog-close]="forms.value"
-      [disabled]="!forms.valid"
+      [mat-dialog-close]="outputData.value"
+      [disabled]="!parts.valid"
     >
       <span [matTooltip]="tooltipOnSave" i18n>Save</span>
```
Comment (TheSlimvReal) [inline src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.html:106]: This getter is not really needed
```suggestion
      <span matTooltip="Fix the errors to save the form" [matTooltipDisabled]="form.valid" i18n-matTooltip="Shown when there are errors that prevent saving" i18n>Save</span>
```

**Review by TheSlimvReal — COMMENTED** [review]
Looks good now, just some small cleanup ideas

File: src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.html [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
-  <h1 i18n="Edit the progress of one or multiple tasks">Edit Progress</h1>
+  <mat-form-field class = "title-field">
+      <input
+        [formControl]="title"
+        matInput
+        type="text"
+        i18n-placeholder="Title Progress|Edit the progress of one or multiple tasks"
+        placeholder="Title"
+      />
+  </mat-form-field>
```
[the remark this answers is not in the document]
Comment (SuttArt) [reply src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.html:10]: Done

File: src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.html (hunk 1 above)
Comment (SuttArt) [reply src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.html:106]: Done

File: src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.ts [hunk 3]
```diff
@@ -70,17 +81,17 @@ export class EditProgressDashboardComponent {
       currentValue: 1,
       targetValue: 10,
     };
-    this.forms.push(this.formGroup(newPart));
+    this.parts.push(this.formGroup(newPart));
   }
 
   get tooltipOnSave(): string {
```
[the remark this answers is not in the document]
Comment (SuttArt) [reply src/app/features/progress-dashboard-widget/edit-progress-dashboard/edit-progress-dashboard.component.ts:87]: Done

**Review by TheSlimvReal — APPROVED** [review]
Good work. Thank you for implementing this feature!

**Pull request merged**